code/modules/mafia/controller.dm

```diff
@@ -839,6 +920,27 @@
 	if(GLOB.mafia_signup.len >= MAFIA_MAX_PLAYER_COUNT || custom_setup.len)//enough people to try and make something (or debug mode)
 		basic_setup()
 
+/**
+  * Filters inactive player into a different list until they reconnect, and removes players who are no longer ghosts.
+  *
+  * If a disconnected player gets a non-ghost mob and reconnects, they will be first put back into mafia_signup then filtered by that.
+  */
+/datum/mafia_controller/proc/check_signups()
+	for(var/bad_key in GLOB.mafia_bad_signup)
+		if(GLOB.directory[bad_key] && GLOB.directory[bad_key] == GLOB.mafia_bad_signup[bad_key])
```

Comment:
After reconnection client won't be the same ?

Reply:
Didn't see this problem come up, maybe because I used directory?

Reply:
NVM, totally see what you mean